client/components/button/docs/example.jsx

```diff
@@ -36,6 +38,22 @@ class Buttons extends React.PureComponent {
 					{ toggleText }
 				</a>
 				{ this.renderButtons( toggleText ) }
+
+				<h3>Block-level Button Wrapper</h3>
+				<Card>
+					<div className="docs__design-button-row">
+						<BlockButton onClick={ noop }>
+							<p>
+								This "button" can contain other elements as children, including <code>p</code>s,{' '}
```

Comment:
I'd recommend adding a _why_ explanation of when we'd use this.

Reply:
Another section that has something like the items in https://github.com/Automattic/wp-calypso/pull/21753 would also give a better idea too